payments.py

```diff
@@ -0,0 +1,40 @@
+from django.contrib.auth.models import Permission, User
+
+from memberships.models import Member, Membership, FailedPayment, Payment
+from funky_time import epoch_to_datetime
+
+
+def handle_stripe_payment(event):
+    if event["type"] == "invoice.payment_failed":
+        FailedPayment.objects.create(
+            stripe_customer_id=event["data"]["object"]["customer"],
+            stripe_subscription_id=event["data"]["object"]["subscription"],
+            stripe_event_type=event["type"],
+        )
+    if event["type"] == "invoice.payment_succeeded":
+        member = Member.objects.get(email=event["data"]["object"]["customer_email"])
```

Comment:
I'm on the fence about this, but it feels like we should handle the case where the Member does not exist. I can think of two cases where it is possible for this to happen, but they should both result in the same exception from `Member.objects.get`:

1. The `customer_email` is empty for some reason (accidentally removed, or removed for GDPR reasons?)
2. A customer with that email address no longer exists (their email has been changed somehow and not updated in Stripe, or their account details have been removed / anonymised for GDPR reasons).

If either of these cases happen I think it is reasonable for us to return an error to Stripe and maybe log something in the failed payments table.

According to https://stripe.com/docs/webhooks/build#return-a-2xx-status-code-quickly if you return a non `2xx` status code stripe will consider this a failure and try to resend, and then eventually give up, which could help iron out any race conditions.

Reply:
Yes, most definitely. Will do.

Reply:
@CarwynNelson how will this relate to #223?

Once there is a primary key in `failed_payment`, what will happen when a payment fails, it goes to be logged, but there is no member for that `stripe_customer_id`?

Reply:
> what will happen when a payment fails, it goes to be logged, but there is no member for that stripe_customer_id?

Good question. We could add a fallback to email a member of the team to look into this, but I think we should first check to see if it's even possible before we do. Or maybe there is something we can do to ensure that it is practically impossible for it to happen.

Reply:
So when a payment is received (successful or failed), which has no member associated with the `stripe_customer_id`, an email should be sent to a member of the team (e.g. support@geek.zone) if there is no way to ensure that it is practically impossible for it to happen?

I think both should be implemented - but not too sure about the first.

Reply:
Is it worth raising this as an issue so we can think about it / mull over it later?

It seems like an unlikely case so it feels like we make forward progress now and come back to it as a tidy up task later.

Make sure to include enough information so that we remember the context later on. I've made the mistake of not doing that too many times :stuck_out_tongue:

Reply:
Okay - this seems too big for this PR. Shall close this an create an issue 👍